### PR TITLE
Give each driver a subcommand

### PR DESCRIPTION
**Pull request by cpuguy83**
Instead of sticking everything into the `create` command, create subcommands for each driver.
Also removes the driver flag's prefixes (which contained the driver name)

- [x] May be an issue with global flags being passed into subcommands codegangsta/cli#191 --- No issue
- [x] ~~Split swarm/engine opts from driver opts in the help output~~ not needed for this, can look at it separately
- [x] Fix help output command format - codegangsta/cli#246

```
# ./machine create --help
NAME:
   machine create - Create a machine

USAGE:
   machine create command [command options] [arguments...]

COMMANDS:
   amazonec2		Create a machine using the amazonec2 driver
   azure		Create a machine using the azure driver
   digitalocean		Create a machine using the digitalocean driver
   exoscale		Create a machine using the exoscale driver
   generic		Create a machine using the generic driver
   google		Create a machine using the google driver
   none			Create a machine using the none driver
   openstack		Create a machine using the openstack driver
   rackspace		Create a machine using the rackspace driver
   softlayer		Create a machine using the softlayer driver
   virtualbox		Create a machine using the virtualbox driver
   vmwarevcloudair	Create a machine using the vmwarevcloudair driver
   vmwarevsphere	Create a machine using the vmwarevsphere driver
   help, h		Shows a list of commands or help for one command

OPTIONS:
   --help, -h	show help
```

```
# ./machine create virtualbox -help
Usage: docker-machine create virtualbox [OPTIONS] [arg...]

Create a machine using the virtualbox driver

Options:

   --boot2docker-url 											The URL of the boot2docker image. Defaults to the latest available version [$VIRTUALBOX_BOOT2DOCKER_URL]
   --cpu-count "1"											number of CPUs for the machine (-1 to use the number of CPUs available) [$VIRTUALBOX_CPU_COUNT]
   --disk-size "20000"											Size of disk for host in MB [$VIRTUALBOX_DISK_SIZE]
   --hostonly-cidr "192.168.99.1/24"									Specify the Host Only CIDR [$VIRTUALBOX_HOSTONLY_CIDR]
   --import-boot2docker-vm 										The name of a Boot2Docker VM to import
   --memory "1024"											Size of memory for host in MB [$VIRTUALBOX_MEMORY_SIZE]
   --engine-install-url "https://get.docker.com"							Custom URL to use for engine installation [$MACHINE_DOCKER_INSTALL_URL]
   --engine-opt [--engine-opt option --engine-opt option]						Specify arbitrary flags to include with the created engine in the form flag=value
   --engine-insecure-registry [--engine-insecure-registry option --engine-insecure-registry option]	Specify insecure registries to allow with the created engine
   --engine-registry-mirror [--engine-registry-mirror option --engine-registry-mirror option]		Specify registry mirrors to use
   --engine-label [--engine-label option --engine-label option]						Specify labels for the created engine
   --engine-storage-driver 										Specify a storage driver to use with the engine
   --swarm												Configure Machine with Swarm
   --swarm-image "swarm:latest"										Specify Docker image to use for Swarm [$MACHINE_SWARM_IMAGE]
   --swarm-master											Configure Machine to be a Swarm master
   --swarm-discovery 											Discovery service to use with Swarm
   --swarm-strategy "spread"										Define a default scheduling strategy for Swarm
   --swarm-opt [--swarm-opt option --swarm-opt option]							Define arbitrary flags for swarm
   --swarm-host "tcp://0.0.0.0:3376"									ip/socket to listen on for Swarm master
   --swarm-addr 											addr to advertise for Swarm (default: detect and use the machine IP)
```

Formatting issue above are just from copy/paste

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @ehazlett @nathanleclaire

File: drivers/vmwarevcloudair/vcloudair.go [hunk 1]
```diff
@@ -50,48 +50,48 @@ func GetCreateFlags() []cli.Flag {
 	return []cli.Flag{
 		cli.StringFlag{
 			EnvVar: "VCLOUDAIR_USERNAME",
-			Name:   "vmwarevcloudair-username",
+			Name:   "-username",
```
Comment (nathanleclaire) [inline drivers/vmwarevcloudair/vcloudair.go]: These want to have a dash in front?  Why is that?

Comment (cpuguy83) [inline drivers/vmwarevcloudair/vcloudair.go]: hah, good catch, bad replace.

Comment (nathanleclaire) [inline drivers/vmwarevcloudair/vcloudair.go]: OK nice I was wondering if I was missing something :)

**Comment by cpuguy83** [comment]
Updated to fix replacement issues

**Comment by nathanleclaire** [comment]
I would love to merge this, but I'm going to close for now -- we've had enough migration issues so far that I'm quite paranoid about breaking backwards compatibility in such a dramatic fashion :(

**Pull request closed without merging**